confmap/provider/googlesecretmanagerprovider/provider_test.go

```diff
@@ -0,0 +1,137 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package googlesecretmanagerprovider
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"cloud.google.com/go/secretmanager/apiv1/secretmanagerpb"
+	gax "github.com/googleapis/gax-go/v2"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/confmap"
+	"google.golang.org/grpc/codes"
+)
+
+var validSecrets = map[string]string{
```

Comment:
It would be a lot more obvious when reading test code below if instead of being a set global variable, the map of secrets was a field on the secrets manager client that was set when a mock client is allocated.

Reply:
Done